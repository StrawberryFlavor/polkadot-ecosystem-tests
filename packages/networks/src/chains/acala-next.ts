import { acala, karura } from './acala.js'

import path from 'path'

export const acalaNext = acala.extend(() => ({
  name: 'acalaNext',
  wasmOverride: path.resolve(__dirname, '../wasm/acala_runtime.wasm'),
}))

export const karuraNext = karura.extend(() => ({
  name: 'karuraNext',
  wasmOverride: path.resolve(__dirname, '../wasm/karura_runtime.wasm'),
}))
