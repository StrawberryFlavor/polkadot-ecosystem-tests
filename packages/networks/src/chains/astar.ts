import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccount.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccount.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'https://rpc.astar.network',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'https://rpc.shiden.astar.network',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
