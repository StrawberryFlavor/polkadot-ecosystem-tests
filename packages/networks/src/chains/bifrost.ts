import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
  },
  bifrost: {
    relayToken: 'KSM',
  },
}

const getInitStorages = (config: typeof custom.bifrostPolkadot | typeof custom.bifrost) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
  Tokens: {
    account: [[[defaultAccount.alice.address, config.relayToken], { balance: 10 * 1e12 }]],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  paraId: 2030,
  endpoint: 'wss://hk.p.bifrost-rpc.liebi.com/ws',
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrost = defineChain({
  name: 'bifrost',
  paraId: 2001,
  endpoint: 'wss://bifrost-rpc.liebi.com/ws',
  custom: custom.bifrost,
  initStorages: getInitStorages(custom.bifrost),
})
