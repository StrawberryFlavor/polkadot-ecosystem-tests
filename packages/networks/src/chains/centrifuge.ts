import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  centrifuge: {
    acalaUsd: 'AUSD',
  },
  altair: {
    acalaUsd: 'KUSD',
  },
}

const getInitStorages = (config: typeof custom.centrifuge | typeof custom.altair) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
  OrmlTokens: {
    accounts: [[[defaultAccount.alice.address, config.acalaUsd], { balance: 10 * 1e12 }]],
  },
})

export const centrifuge = defineChain({
  name: 'centrifuge',
  paraId: 2031,
  endpoint: 'wss://fullnode.centrifuge.io',
  custom: custom.centrifuge,
  initStorages: getInitStorages(custom.centrifuge),
})

export const altair = defineChain({
  name: 'altair',
  paraId: 2088,
  endpoint: 'wss://fullnode.altair.centrifuge.io',
  custom: custom.altair,
  initStorages: getInitStorages(custom.altair),
})
