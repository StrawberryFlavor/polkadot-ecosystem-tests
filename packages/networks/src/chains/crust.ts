import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  crustPolkadot: {},
  crust: {},
}

const getInitStorages = () => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 1, data: { free: '1000000000000000' } }]],
  },
})

export const crustPolkadot = defineChain({
  name: 'crustPolkadot',
  paraId: 2008,
  endpoint: 'wss://crust-parachain.crustapps.net',
  custom: custom.crustPolkadot,
  initStorages: getInitStorages(),
})

export const crust = defineChain({
  name: 'crust',
  paraId: 2012,
  endpoint: 'wss://rpc-shadow.crust.network/',
  custom: custom.crust,
  initStorages: getInitStorages(),
})
