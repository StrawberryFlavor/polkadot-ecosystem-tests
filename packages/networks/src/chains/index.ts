export * from './acala-next.js'
export { acala, karura } from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
