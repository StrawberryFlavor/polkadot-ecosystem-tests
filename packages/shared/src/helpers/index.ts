import { expect } from 'vitest'
import { testingPairs, withExpect } from '@acala-network/chopsticks-testing'

export const defaultAccount = testingPairs()

export type DefaultAccount = ReturnType<typeof testingPairs>

const { check, checkEvents, checkHrmp, checkSystemEvents, checkUmp } = withExpect((x: any) => ({
  toMatchSnapshot(msg?: string): void {
    expect(x).toMatchSnapshot(msg)
  },
  toMatch(value: any, _msg?: string): void {
    expect(x).toMatch(value)
  },
  toMatchObject(value: any, _msg?: string): void {
    expect(x).toMatchObject(value)
  },
}))

export { check, checkEvents, checkHrmp, checkSystemEvents, checkUmp }

export * from '@acala-network/chopsticks-testing'
export * from './types.js'
