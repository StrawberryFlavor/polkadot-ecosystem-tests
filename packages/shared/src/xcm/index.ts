export * from './runXcmPalletDown.js'
export * from './runXcmPalletHorizontal.js'
export * from './runXtokensUp.js'
export * from './runXtokenstHorizontal.js'
