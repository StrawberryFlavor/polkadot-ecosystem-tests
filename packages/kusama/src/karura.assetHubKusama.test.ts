import { afterAll, describe } from 'vitest'
import { connectParachains } from '@acala-network/chopsticks'
import { defaultAccount } from '@e2e-test/shared/helpers'

import { assetHubKusama, karura } from '@e2e-test/networks/chains'
import { createNetwork } from '@e2e-test/networks'
import { query, tx } from '@e2e-test/shared/api'
import { runXcmPalletHorizontal } from '@e2e-test/shared/xcm'

// assetHubKusama <=> karura
describe(`'assetHubKusama' <-> 'karura' xcm transfer 'USDT'`, async () => {
  const [assetHubKusamaClient, karuraClient] = await Promise.all([createNetwork(assetHubKusama), createNetwork(karura)])

  await connectParachains([assetHubKusamaClient.chain, karuraClient.chain])

  const assetHubKusamaUsdt = assetHubKusamaClient.config.custom!.usdtIndex
  const karuraUsdt = karuraClient.config.custom!.usdt

  afterAll(async () => {
    await assetHubKusamaClient.teardown()
    await karuraClient.teardown()
  })

  runXcmPalletHorizontal(`'assetHubKusama' -> 'karura' USDT`, async () => {
    return {
      fromChain: assetHubKusamaClient,
      toChain: karuraClient,
      fromBalance: query.assets(assetHubKusamaUsdt),
      toBalance: query.tokens(karuraUsdt),
      tx: tx.xcmPallet.limitedReserveTransferAssetsV3(
        assetHubKusamaClient.config.custom!.usdt,
        1e6,
        tx.xcmPallet.parachainV3(1, karuraClient.config.paraId!),
      ),
    }
  })

  runXcmPalletHorizontal(`'karura' -> 'assetHubKusama' USDT`, async () => {
    await karuraClient.dev.setStorage({
      Tokens: {
        Accounts: [[[defaultAccount.alice.address, karuraUsdt], { free: 10e6 }]],
      },
    })

    return {
      fromChain: karuraClient,
      toChain: assetHubKusamaClient,
      fromAccount: defaultAccount.alice,
      toAccount: defaultAccount.alice,
      fromBalance: query.tokens(karuraUsdt),
      toBalance: query.assets(assetHubKusamaUsdt),
      tx: tx.xtokens.transfer(karuraUsdt, 1e6, tx.xtokens.parachainV3(assetHubKusamaClient.config.paraId!)),
    }
  })
})
