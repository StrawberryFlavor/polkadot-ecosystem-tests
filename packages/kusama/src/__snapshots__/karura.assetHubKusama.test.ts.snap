// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > balance on from chain 1`] = `
{
  "balance": 999000000,
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > balance on from chain 2`] = `
{
  "balance": 998000000,
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > balance on from chain 3`] = `
{
  "balance": 997000000,
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > balance on to chain 1`] = `
{
  "free": "(rounded 999000)",
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > balance on to chain 2`] = `
{
  "free": "(rounded 3000000)",
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > from chain hrmp messages 1`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v3": [
          {
            "reserveAssetDeposited": [
              {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "allCounted": 1,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": null,
                    },
                  },
                },
                "parents": 0,
              },
            },
          },
          {
            "setTopic": "0x1321f7f4242fc20acb5e33f3003f4c5db8aeee76bb280db7a518b502ba1b316b",
          },
        ],
      },
    ],
    "recipient": 2000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > from chain hrmp messages 2`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v3": [
          {
            "reserveAssetDeposited": [
              {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "allCounted": 1,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": null,
                    },
                  },
                },
                "parents": 0,
              },
            },
          },
          {
            "setTopic": "0x7257053364dc18e4d0358e2650b1e486c0f189c3f80bf5d0b06b98d1139705bc",
          },
        ],
      },
    ],
    "recipient": 2000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > from chain hrmp messages 3`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v3": [
          {
            "reserveAssetDeposited": [
              {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "allCounted": 1,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": null,
                    },
                  },
                },
                "parents": 0,
              },
            },
          },
          {
            "setTopic": "0xa81ef18b74005fa977d4069a5d6004c47d0ba917f8817852e28ee24bcdee3f0b",
          },
        ],
      },
    ],
    "recipient": 2000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > to chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Sibling": 1000,
      },
      "success": true,
      "weightUsed": {
        "proofSize": 0,
        "refTime": 1000000000,
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > tx events 1`] = `[]`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > tx events 2`] = `[]`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'assetHubKusama' -> 'karura' USDT > tx events 3`] = `[]`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on from chain 1`] = `
{
  "free": 9000000,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on from chain 2`] = `
{
  "free": 9000000,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on from chain 3`] = `
{
  "free": 9000000,
  "frozen": 0,
  "reserved": 0,
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on to chain 1`] = `
{
  "balance": "(rounded 1000000000)",
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on to chain 2`] = `
{
  "balance": "(rounded 999000000)",
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > balance on to chain 3`] = `
{
  "balance": "(rounded 1000000000)",
  "extra": null,
  "reason": {
    "consumer": null,
  },
  "status": "Liquid",
}
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > from chain hrmp messages 1`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v2": [
          {
            "withdrawAsset": [
              {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x2": [
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 0,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x2": [
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 0,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "all": null,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": {
                        "any": null,
                      },
                    },
                  },
                },
                "parents": 0,
              },
              "maxAssets": 1,
            },
          },
        ],
      },
    ],
    "recipient": 1000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > from chain hrmp messages 2`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v2": [
          {
            "subscribeVersion": {
              "maxResponseWeight": 0,
              "queryId": 31,
            },
          },
        ],
      },
    ],
    "recipient": 1000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > from chain hrmp messages 3`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v2": [
          {
            "withdrawAsset": [
              {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x2": [
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 0,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 1000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x2": [
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 0,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "all": null,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": {
                        "any": null,
                      },
                    },
                  },
                },
                "parents": 0,
              },
              "maxAssets": 1,
            },
          },
        ],
      },
    ],
    "recipient": 1000,
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > to chain xcm events 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Sibling": 2000,
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 7200)",
        "refTime": "(rounded 320000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > to chain xcm events 2`] = `
[
  {
    "data": {
      "id": "(hash)",
      "origin": {
        "Sibling": 2000,
      },
      "success": true,
      "weightUsed": {
        "proofSize": "(rounded 7200)",
        "refTime": "(rounded 320000000)",
      },
    },
    "method": "Processed",
    "section": "messageQueue",
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > tx events 1`] = `
[
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": 1000000,
          },
          "id": {
            "interior": {
              "X3": [
                {
                  "Parachain": 1000,
                },
                {
                  "PalletInstance": 50,
                },
                {
                  "GeneralIndex": "(rounded 2000)",
                },
              ],
            },
            "parents": 1,
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": 1000,
            },
            {
              "AccountId32": {
                "id": "(hash)",
                "network": null,
              },
            },
          ],
        },
        "parents": 1,
      },
      "fee": {
        "fun": {
          "Fungible": 1000000,
        },
        "id": {
          "interior": {
            "X3": [
              {
                "Parachain": 1000,
              },
              {
                "PalletInstance": 50,
              },
              {
                "GeneralIndex": "(rounded 2000)",
              },
            ],
          },
          "parents": 1,
        },
      },
      "sender": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
    },
    "method": "TransferredAssets",
    "section": "xTokens",
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > tx events 2`] = `
[
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": 1000000,
          },
          "id": {
            "interior": {
              "X3": [
                {
                  "Parachain": 1000,
                },
                {
                  "PalletInstance": 50,
                },
                {
                  "GeneralIndex": "(rounded 2000)",
                },
              ],
            },
            "parents": 1,
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": 1000,
            },
            {
              "AccountId32": {
                "id": "(hash)",
                "network": null,
              },
            },
          ],
        },
        "parents": 1,
      },
      "fee": {
        "fun": {
          "Fungible": 1000000,
        },
        "id": {
          "interior": {
            "X3": [
              {
                "Parachain": 1000,
              },
              {
                "PalletInstance": 50,
              },
              {
                "GeneralIndex": "(rounded 2000)",
              },
            ],
          },
          "parents": 1,
        },
      },
      "sender": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
    },
    "method": "TransferredAssets",
    "section": "xTokens",
  },
]
`;

exports[`'assetHubKusama' <-> 'karura' xcm transfer 'USDT' > 'karura' -> 'assetHubKusama' USDT > tx events 3`] = `
[
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": 1000000,
          },
          "id": {
            "interior": {
              "X3": [
                {
                  "Parachain": 1000,
                },
                {
                  "PalletInstance": 50,
                },
                {
                  "GeneralIndex": "(rounded 2000)",
                },
              ],
            },
            "parents": 1,
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": 1000,
            },
            {
              "AccountId32": {
                "id": "(hash)",
                "network": null,
              },
            },
          ],
        },
        "parents": 1,
      },
      "fee": {
        "fun": {
          "Fungible": 1000000,
        },
        "id": {
          "interior": {
            "X3": [
              {
                "Parachain": 1000,
              },
              {
                "PalletInstance": 50,
              },
              {
                "GeneralIndex": "(rounded 2000)",
              },
            ],
          },
          "parents": 1,
        },
      },
      "sender": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
    },
    "method": "TransferredAssets",
    "section": "xTokens",
  },
]
`;
